File: low-level-design/design-patterns/python/behavioural/chain-of-responsibility-pattern/deployment_service.py
from handler import Handler,DeploymentRequest

class DeploymentService:
    def __init__(self, handler: Handler):
        self.handler = handler

    def deploy(self, request: DeploymentRequest):
        print(f"DeploymentService: Attempting to deploy {request.service_name}...")
        if self.handler.handle(request):
            print(f"DeploymentService: Successfully deployed {request.service_name}.")
            return True
        else:
            print(f"DeploymentService: Failed to deploy {request.service_name}.")
            return False

File: low-level-design/design-patterns/golang/behavioural/chain-of-responsibility-pattern/cor/deployment_service.go
package cor

func NewDeploymentService(handler Handler) *DeploymentService {
	return &DeploymentService{
		handler: handler,
	}
}

type DeploymentService struct {
	handler Handler
}

func (h *DeploymentService) Deploy(request *DeploymentRequest) error {
	return h.handler.Handle(request)
}
